src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,39 @@
+package duke.task;
+
+import java.util.Date;
+
+public abstract class Task {
+
+    final String description;
+    boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public abstract String toEncodedString();
```

Comment:
Your tasks are responsible for encoding themselves but there's a separate parser class.